src/firmware/app/control/trajectory_planner.h

```diff
@@ -152,27 +124,87 @@ app_trajectory_planner_interpolateConstantPeriodPositionTrajectory(
     PositionTrajectory_t* variable_time_trajectory, const float interpolation_period);
 
 /**
- * Generates the X/Y points for each position on the constant arc length trajectory
+ * Builds the forwards continuous linear velocity profile for the given parameters.
+ * NOTE: The velocity profile returned by this function is NOT reverse continuous and does
+ * not guarantee a feasible path to follow.
+ *
+ * NOTE: The velocity_profile will contain the same number of elements as the input
+ * max_allowable_linear_speed[] array
+ *
+ * @pre The velocity_profile array is pre-allocated to contain up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS elements
+ *
+ * @param trajectory [in/out] The trajectory that contains the path parameters and will be
+ * modified max_allowable_speed_profile input array
+ *
+ * @param max_allowable_speed_profile [in] The pre-allocated array that will contain all
+ * of the maximum allowable velocity values for each point on the curve
+ *
+ * @param segment_lengths [in] The length of each corresponding segment [meters]
+ *
+ * @return The generation status. Can be OK or an error message
+ * */
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_createForwardsContinuousLinearSpeedProfile(
+    PositionTrajectory_t* trajectory, float* max_allowable_speed_profile,
+    TrajectorySegment_t* segment_lengths);
+
+/**
+ * Builds the forwards continuous linear velocity profile for the given parameters.
+ * NOTE: The velocity profile returned by this function is NOT reverse continuous and does
+ * not guarantee a feasible path to follow.
+ *
+ * NOTE: The velocity_profile will contain the same number of elements as the input
+ * max_allowable_linear_speed[] array
```

Comment:
I'm confused where `max_allowable_linear_speed` is coming from.

Reply:
Removed references to max_allowable_linear_speed. It was copy-pasted from the old version 👍